sdks/python/apache_beam/transforms/external.py

```diff
@@ -652,7 +652,7 @@ class JavaJarExpansionService(object):
   """
   def __init__(self, path_to_jar, extra_args=None):
     if extra_args is None:
-      extra_args = ['{{PORT}}']
+      extra_args = ['{{PORT}}', f'--filesToStage={path_to_jar}']
```

Comment:
This is broken when `path_to_jar` is anything but a local file. I noticed this during 2.34.0RC1 validation, when I usually test xlang SQL by specifying `--beam_services '{":sdks:java:extensions:sql:expansion-service:shadowJar": "http://staged/sql/expansionservice.jar"}'`.

I think it may also be broken for the final release. Won't `path_to_jar` be `:sdks:java:extensions:sql:expansion-service:shadowJar` in that case?

Reply:
I think we need to add this to extra_args in __enter__, after self._path_to_jar has been resolved to a local file.

Reply:
Ah actually I think the gradle target will get resolved to a URL (in `BeamJarExpansionService.__init__`), but we'll pass the URL in filesToStage, which won't work.

Reply:
Thanks. Agree. I think this is a definite regression for x-lang.

Reply:
Good point. I'll create a fix.

Reply:
I just cerated a fix here: https://github.com/apache/beam/pull/15889

Reply:
https://github.com/apache/beam/pull/15891

(FWIW, it did handle gradle path resolution, which was tested, but just not remote http://staged/sql/expansionservice.jar, i.e. the `JavaJarServer.local_jar` part.)

Reply:
Ah, well if it handles gradle path resolution it doesn't necessarily need to block the release.

Reply:
I think Gradle target resolves to a URL: https://github.com/apache/beam/blob/0faa3b9403bd9c852075f2baf9b2e1f9d588677f/sdks/python/apache_beam/utils/subprocess_server.py#L189

So it will still fail.

Reply:
Good point. (In a dev branch it resolves to a local file, but that's a special case...)